SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityOfferScoring.java

```diff
@@ -0,0 +1,288 @@
+package com.hubspot.singularity.mesos;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.hubspot.mesos.Resources;
+import com.hubspot.singularity.AgentMatchState;
+import com.hubspot.singularity.RequestUtilization;
+import com.hubspot.singularity.SingularityAgentUsage;
+import com.hubspot.singularity.SingularityDeployStatistics;
+import com.hubspot.singularity.SingularityPendingTaskId;
+import com.hubspot.singularity.SingularityTask;
+import com.hubspot.singularity.SingularityTaskId;
+import com.hubspot.singularity.SingularityTaskRequest;
+import com.hubspot.singularity.config.MesosConfiguration;
+import com.hubspot.singularity.config.SingularityConfiguration;
+import com.hubspot.singularity.data.DeployManager;
+import com.hubspot.singularity.data.TaskManager;
+import com.hubspot.singularity.data.usage.UsageManager;
+import com.hubspot.singularity.helpers.MesosUtils;
+import com.hubspot.singularity.mesos.SingularityAgentUsageWithCalculatedScores.MaxProbableUsage;
+import java.util.Map;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/*
+ * It is assumed that all code in this class is run behind a request-level lock
```

Comment:
Is there a "checkRequestLockOwner" or similar method that could be used by this class to make violations of this assumption more explicit / easier to debug?